op_builder/cpu_adagrad.py

```diff
@@ -38,10 +38,4 @@ def include_paths(self):
             CUDA_INCLUDE = []
         elif not self.is_rocm_pytorch():
             CUDA_INCLUDE = [os.path.join(torch.utils.cpp_extension.CUDA_HOME, "include")]
-        else:
```

Comment:
@rraminen @jithunnair-amd @BLOrange-AMD - this is a re-implementation of PR #4172.  Since I'm not able to re-run the tests on that PR due to the permissions on the ROCm fork that it is from.  Do these changes make sense and should we go ahead with merging these changes?

Reply:
Looks like CUDA_INCLUDE is not initialized with ROCm. Original PR has this: https://github.com/ROCmSoftwarePlatform/DeepSpeed/blob/00120efcc7796d302c395f6d1f0e9007335ea5c1/op_builder/cpu_adagrad.py#L42

Reply:
@loadams "Since I'm not able to re-run the tests on that PR due to the permissions on the ROCm fork that it is from" ... Can you please elaborate what we need to change on the permissions for our fork, so we can enable CI running on PRs from our fork?

Reply:
@BLOrange-AMD - good catch, thanks.  I'm also happy to merge your original PR, just I'm not able to re-run the transient test/merge the latest changes into it.  If you could do that we can just merge that one?

Reply:
@loadams Since you're able to kick off CI on MI200 using this PR, let's use this PR as the one to target merge. I wouldn't want to merge a PR that we cannot run CI on.

Reply:
I've not extensively tested these on MI200, though I can.  But the issue is I'm not even able to merge the other PR as there is a legacy failing test that was transient, but I cannot merge master into it.  Normally there is a button here that allows me to update the PR

![image](https://github.com/microsoft/DeepSpeed/assets/114770087/a6ff33ce-3e3b-43af-a440-af997287cae8)


vs a PR that has this:

![image](https://github.com/microsoft/DeepSpeed/assets/114770087/c5c87cde-cec8-4cc5-ba13-af9d50b68179)

Reply:
Running on MI200 here: https://github.com/microsoft/DeepSpeed/actions/runs/6567468508/job/17840231886

Reply:
Weirdly that was failing while installing apex, so I ran again with it also using the branch version: https://github.com/microsoft/DeepSpeed/actions/runs/6567671904/job/17840802181

Reply:
It appears to have failed due to existing issues in our CI, but it is running the tests as expected, so there is no impact on the pre-ROCm 5 stuff.

Reply:
@jithunnair-amd - running [here](https://github.com/microsoft/DeepSpeed/actions/runs/6589976400/job/17905603326) on updated PR - let me know if this looks good to merge?

Reply:
The above run failed with "5 failed, 42 passed, 10 skipped, 16 warnings", whereas [one of your earlier runs](https://github.com/microsoft/DeepSpeed/actions/runs/6565890180/job/17835413021) with the apex fixes ran many more tests: `79 failed, 735 passed, 129 skipped, 40 warnings`. That looks like the failures in the above job just terminated the run?

Reply:
Yes, @jithunnair-amd - that's due to the environment issues on our side we are still trying to work out to get resolved.

Reply:
But if the overall PR is fine (and doesn't appear to be causing these from our side), then we should be fine to go ahead and complete the PR?

Reply:
Yes, sounds good! 

Btw, we have taken an internal action item to look at the DeepSpeed unit test status upstream so we can get this workflow to green status. If there are certain failures that you know are due to CI environment issues, please let us know so we can exclude them from our investigation.